node_modules/oae-core/setpermissions/js/setpermissions.js

```diff
@@ -63,43 +63,35 @@ define(['jquery', 'oae.core'], function($, oae) {
             // Get the items in the share with list
             var selection = oae.api.util.autoSuggest().getSelection($rootel);
 
-            var folders = [];
-            var shared = [];
-
-            $.each(selection, function(index, selectedItem) {
-                // All folders need to be extracted from the selection list as they need to be
-                // passed in separately
-                if (selectedItem.resourceType === 'folder') {
-                    folders.push(selectedItem);
-                // All other items that are different from the current user can be added to
-                // the list of users and people the item will be shared with
-                } else {
-                    shared.push(selectedItem);
-                }
+            var selectedFolderAndPrincipalItems = _.partition(selection, function(selectedItem) {
```

Comment:
You think that's cool? Just wait till we can use extraction with it:

`var [folderItems, principalItems] = _.partition(selection, function(...) {`

So much nicer.